src/StateService/StatePlugin.cs

```diff
@@ -233,5 +235,69 @@ public JObject GetStateHeight(JArray _params)
             json["validatedrootindex"] = StateStore.Singleton.ValidatedRootIndex;
             return json;
         }
+
+        private ContractState GetHistoricalContractState(MPTTrie<StorageKey, StorageItem> trie, UInt160 script_hash)
+        {
+            const byte prefix = 8;
+            StorageKey skey = new KeyBuilder(NativeContract.ContractManagement.Id, prefix).Add(script_hash);
+            return trie[skey]?.GetInteroperable<ContractState>();
+        }
+
+        [RpcMethod]
+        public JObject GetState(JArray _params)
+        {
+            if (!uint.TryParse(_params[0].AsString(), out uint index))
+            {
+                if (!UInt256.TryParse(_params[0].AsString(), out UInt256 root_hash))
+                    throw new RpcException(-100, "Invalid block hash or index");
+                var header = NativeContract.Ledger.GetHeader(System.StoreView, root_hash);
+                if (header is null)
+                    throw new RpcException(-100, "Unkown block");
+                index = header.Index;
+            }
+            var script_hash = UInt160.Parse(_params[1].AsString());
+            var key = Convert.FromBase64String(_params[2].AsString());
+            var isFind = _params.Count > 3 && _params[3].AsBoolean();
```

Comment:
It seems like this should be two separate Rpc Methods - one for get and one for find - instead of using a parameter to change behavior

Reply:
Done.